barretenberg/cpp/src/barretenberg/client_ivc/client_ivc.test.cpp

```diff
@@ -125,12 +125,13 @@ TEST_F(ClientIVCTests, Full)
     auto intermediary_acc = update_accumulator_and_decide_native(
         ivc.prover_fold_output.accumulator, kernel_fold_proof, foo_verifier_instance, kernel_vk);
 
-    VerifierFoldData kernel_fold_output = { kernel_fold_proof, function_vk_with_merge };
+    VerifierFoldData kernel_fold_output = { kernel_fold_proof, kernel_vk };
     size_t NUM_CIRCUITS = 1;
     for (size_t circuit_idx = 0; circuit_idx < NUM_CIRCUITS; ++circuit_idx) {
         // Accumulate function circuit
         Builder function_circuit = create_mock_circuit(ivc);
         FoldProof function_fold_proof = ivc.accumulate(function_circuit);
+        function_vk_with_merge = std::make_shared<VerificationKey>(ivc.prover_instance->proving_key);
```

Comment:
we used to never reset function_vk_with_merge, so we were using the same vk for all function circuits, which worked generally since all of our function circuits were the same except for the public inputs and the witness values

Reply:
I believe there is no need to reset the vk? the witness values and public inputs are not relevant for the vk. Am I missing something?

Reply:
oh, right, the VerificationKey contains the public inputs.. but still the number of those should not change across this function, hmm.. Also are we certain we want the PIs to be part of the VK?